### DISCUSSION TITLE
Represent `center_mode` as Mercator projected coordinates?

### DISCUSSION THREAD

**Discussion by dwuertz** [podusowski/walkers#69] [Ideas]
While digging through the code, I got the impression that there is a lot of back-and-forth transformations between Mercator / screen coordinates and geodetic coordinates happening during each frame update. This is because, currently, the widget represents its internal state as geodetic coordinates, most prominently, `MapMemory::center_mode`. I propose to switch to normalized (i.e. zoom-independent) Mercator coordinates here instead. My expectation is that this would simplify the code overall and also improve performance / accuracy.

What do you think?

**Comment by podusowski** [comment]
It does sound interesting. I need to put some deeper thought though. I wonder how will it play with the far future plans that I have, namely 3d rendering of the globe, and more provider types, like WMS, which do not necessarily use the Mercator.

I also feel like these math code is generally duplicated between tile drawing functions and the `Projector` struct. On top of that, after your changes, we probably do not need the `AdjustedPosition` anymore. Basically, what I am trying to say here, is that I would like to do some of these first, and see how the code will look like afterwards.
